hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTable.java

```diff
@@ -733,10 +733,11 @@ public HoodieEngineContext getContext() {
   /**
    * Get Table metadata writer.
    *
+   * @param inFlightInstantTimestamp - InFlight instant timestamp for which metadata writer is needed
```

Comment:
rename `inFlightInstantTimestamp` to `triggeringInstantTime` here and metadata writer classes (not timestamp, its mean as logical time, our tests use 001, 002 and everything still works. )

Reply:
`triggeringInstantTime - the instant that is triggering this metadata write`

Reply:
Also when we add ability to index async, this needs to be a `Option<String>` since there will be no inflight/triggering instant then. We can do it here or defer to me. Just calling out